### ISSUE TITLE
Stacking to average Bayesian predictive distributions

### ISSUE THREAD

**Issue by paul-buerkner** [paul-buerkner/brms#268]
The idea is described in http://www.stat.columbia.edu/~gelman/research/unpublished/combine_predictive.pdf and an implementation is provided in dev version of the **loo** package: https://github.com/stan-dev/loo/blob/master/R/bb-stacking.R

There is not much to be done from the **brms** side I guess except to write some wrappers for `model_weights` and `model_select`.

**Comment by sdaza** [comment]
That would be great. 

I am trying to get the `log_lik` from the brms class without success.  

```
library(brms)
library(loo)
library(rstan)

log_lik_list = list()
log_lik_list[[1]] =  extract(mymodel1)[["log_lik"]] 
log_lik_list[[2]] =  extract(mymodel2)[["log_lik"]] 
model_weights_1 = model_weights(log_lik_list, method="stacking")
```

Once I get the weights I have to multiple model predictions by the vector of weights.

**Comment by paul-buerkner** [comment]
Use method log_lik to extract the likelihood.

Sebastian Daza <notifications@github.com> schrieb am Do., 25. Jan. 2018,
20:27:

**Comment by paul-buerkner** [comment]
Sorry, I didn't see you were already trying this method. What is it that doesn't work for you?

**Comment by sdaza** [comment]
The log_lik method works perfectly!
Thanks!

**Comment by paul-buerkner** [comment]
The required wrappers are implemented in the `new-loo` branch, which will be merged as soon as loo 2.0 is on CRAN.
